src/framework/vst/internal/synth/vstsynthesiser.cpp

```diff
@@ -202,6 +203,10 @@ samples_t VstSynthesiser::process(float* buffer, samples_t samplesPerChannel)
 
     const msecs_t nextMsecs = samplesToMsecs(samplesPerChannel, m_sampleRate);
     const VstSequencer::EventSequenceMap sequences = m_sequencer.movePlaybackForward(nextMsecs);
+
+    // TODO: this is not a good way of doing this, please comment!
+    m_vstAudioClient->setPlaybackPosition(m_sequencer.playbackPosition());

```

Comment:
You can add the playback position as an argument to VstAudioClient::process:

`muse::audio::samples_t process(float* output, muse::audio::samples_t samplesPerChannel, muse::audio::msecs_t playbackPosition = 0);`

That would be a little simpler